extensions/applicationinsights-dependencies-js/src/ajaxRecord.ts

```diff
@@ -1,83 +1,351 @@
+        let self = this;
+        self.openDone = false;
+        self.setRequestHeaderDone = false;
+        self.sendDone = false;
+        self.abortDone = false;
+    
+        // <summary>True, if onreadyStateChangeCallback function attached to xhr, otherwise false</summary>
+        self.stateChangeAttached = false;
+    }
 }
 
 export class ajaxRecord {
-    public completed = false;
-    public requestHeadersSize = null;
-    public requestHeaders = null;
-    public ttfb = null;
-    public responseReceivingDuration = null;
-    public callbackDuration = null;
-    public ajaxTotalDuration = null;
-    public aborted = null;
-    public pageUrl = null;
-    public requestUrl = null;
-    public requestSize = 0;
-    public method = null;
+    public completed:boolean;
+    public requestHeadersSize:number;
+    public requestHeaders:any;
+    public responseReceivingDuration:number;
+    public callbackDuration:number;
+    public ajaxTotalDuration:number;
+    public aborted:number;
+    public pageUrl:string;
+    public requestUrl:string;
+    public requestSize:number;
+    public method:string;
+    public perfMark:PerformanceMark;
+    public perfTiming:PerformanceResourceTiming;
+    public perfAttempts?:number;
+    public async?:boolean;
 
     /// <summary>Returns the HTTP status code.</summary>
-    public status = null;
+    public status:string|number;
 
     // <summary>The timestamp when open method was invoked</summary>
-    public requestSentTime = null;
+    public requestSentTime;
 
     // <summary>The timestamps when first byte was received</summary>
-    public responseStartedTime = null;
+    public responseStartedTime;
 
     // <summary>The timestamp when last byte was received</summary>
-    public responseFinishedTime = null;
+    public responseFinishedTime;
 
     // <summary>The timestamp when onreadystatechange callback in readyState 4 finished</summary>
-    public callbackFinishedTime = null;
+    public callbackFinishedTime;
 
     // <summary>The timestamp at which ajax was ended</summary>
-    public endTime = null;
-
-    // <summary>The original xhr onreadystatechange event</summary>
-    public originalOnreadystatechage = null;
+    public endTime;
 
-    public xhrMonitoringState: XHRMonitoringState = new XHRMonitoringState();
+    public xhrMonitoringState: XHRMonitoringState;
 
     // <summary>Determines whether or not JavaScript exception occured in xhr.onreadystatechange code. 1 if occured, otherwise 0.</summary>
-    public clientFailure = 0;
-
+    public clientFailure;
 
     public traceID: string;
     public spanID: string;
 
-    private _logger: IDiagnosticLogger;
-
-    constructor(traceID: string, spanID: string, logger: IDiagnosticLogger) {
-        this.traceID = traceID;
-        this.spanID = spanID;
-        this._logger = logger;
+    public getAbsoluteUrl(): string {
+        // @DynamicProtoStub -- DO NOT add any code as this will be removed during packaging
```

Comment:
Adding so that TypeScript will generate a d.ts file the same as previously, however, the Rush plugin will remove the generated stub during packaging.

e.g. TS will create the function 
ajaxRecord.prototype.getAbsoluteUrl = function() { ... }

But we will remove as this is just code bloat as the entire name can't be minified here, but by defining in the constructor and using dynamicProto() we remove this excessive "ajaxRecord.prototype." but we still end up with a class that defines the "getAbsoluteUrl" function on the ajaxRecord prototype